.github/workflows/ci.yml

```diff
@@ -0,0 +1,31 @@
+name: Clojure CI
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+    branches: ["master"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies
+        run: lein deps
```

Comment:
Unnecessary as all calls to `lein` will resolve deps first.

Reply:
Interesting. It was like that in the default Clojure workflow and I think all the ones I saw during my research. Removing this step!

Reply:
@NoahTheDuke people might do this so as not to get the dependency spam when running other steps:
![image](https://user-images.githubusercontent.com/32344/182222958-43e2619b-099d-48a3-be34-5aa20690d967.png)

I'd suggest keeping it separate even if not strictly necessary to make it easier to troubleshoot.

Reply:
It looks like the cache is only being set when the tests don't fail, so maybe include the fix in this PR to see what happens.